pkg/container/resolver.go

```diff
@@ -100,6 +102,37 @@ func (r *asyncResolver) Finish() ([]Spec, error) {
 	return specs, nil
 }
 
+func (r *asyncResolver) Resolve(spec SourceSpec) (Spec, error) {
+	r.Add(spec)
+	resSlice, err := r.Finish()
+	if err != nil {
+		return Spec{}, err
+	}
+
+	if len(resSlice) != 1 {
+		return Spec{}, fmt.Errorf("unexpected results in container resolver: expected 1 but received %d", len(resSlice))
+	}
+
+	return resSlice[0], nil
+}
+
+func (r *asyncResolver) ResolveAll(sources map[string][]SourceSpec) (map[string][]Spec, error) {
+	containers := make(map[string][]Spec, len(sources))
+	for name, srcList := range sources {
```

Comment:
Just a generic warning - map order when using `range` is not guaranteed in Go. I do not know if that is a problem tho, but I would probably drop a comment that sources can be resolved in random order. Hopefully, this will not change manifest JSON output but theoretically it could (e.g. we upgrade Go version, new map implementation is in place and some output in manifest goes into ordered slice).